src/daemon/common.c

```diff
@@ -361,13 +361,14 @@ int strjoin (char *buffer, size_t buffer_size,
 			avail -= sep_len;
 		}
 
-		field_len = strlen (fields[i]);
-		if (avail < field_len)
+		if (avail == 0)
 			return (-1);
 
-		memcpy (ptr, fields[i], field_len);
+		field_len = strlen (fields[i]);
+
+		memcpy (ptr, fields[i], ((avail < field_len) ? avail : field_len));
 		ptr += field_len;
```

Comment:
Should ptr be incremented by the length actually copied?

So perhaps we could have:

```
field_len = strlen (fields[i]);
copy_len = ((avail < field_len) ? avail : field_len);
memcpy (ptr, fields[i], copy_len);
ptr += copy_len;
avail -= copy_len;
```